ui/lib/kv/addon/components/page/secret/metadata-details.hbs

```diff
@@ -7,6 +7,59 @@
   </:tabLinks>
 
   <:toolbarActions>
-    <ToolbarLink @route="secret.metadata.edit">Edit metadata</ToolbarLink>
+    {{! You must have update on metadata, create is not enough to edit. }}
+    {{#if @model.canUpdateMetadata}}
+      <ToolbarLink @route="secret.metadata.edit" data-test-edit-metadata>Edit metadata</ToolbarLink>
+    {{/if}}
   </:toolbarActions>
-</KvPageHeader>
\ No newline at end of file
+</KvPageHeader>
+
+<div class="form-section">
```

Comment:
Since this isn't a form can the css be updated? Not sure we even need the extra div 🤔 We could do something like what we did in PKI:
```
    <h2 class="title is-4 has-bottom-margin-xs has-top-margin-xl has-border-bottom-light has-bottom-padding-s">
      Cluster Config
    </h2>

Reply:
A little playing with the css, but I got it to look exactly as it was before (okay .5 px smaller on top and bottom, but I'm going with that's okay).
`title is-5 has-bottom-padding-s has-top-margin-l`
Now:
![image](https://github.com/hashicorp/vault/assets/6618863/b6e24c47-6857-48c2-946f-ba1be72d18a1)
Before—old engine:
![image](https://github.com/hashicorp/vault/assets/6618863/a3dad79e-f3e9-4864-9cf2-007e4b7d3964)